src/index.ts

```diff
@@ -143,7 +155,8 @@ function typeUnionConditions(
   project: Project,
   path: string,
   arrayDepth: number,
-  debug: boolean
+  records: IRecord[],
```

Comment:
Not sure how anal I should be about this, but if the array isn't modified the parameter should really be a `ReadonlyArray<IRecord>`/`readonly IRecord[]`.